format/Schema.fbs

```diff
@@ -44,6 +44,36 @@ table FixedSizeList {
   listSize: int;
 }
 
+/// A Map is a logical nested type that is represented as
+///
+/// List<entry: Struct<key: K, value: V>>
+///
+/// In this layout, the keys and values are each respectively contiguous. We do
+/// not constrain the key and value types, so the application is responsible
+/// for ensuring that the keys are hashable and unique. Whether the keys are sorted
+/// may be set in the metadata for this field
+///
+/// In a Field with Map type, the Field has a child Struct field, which then
+/// has two children: key type and the second the value type. The names of the
+/// child fields may be respectively "entry", "key", and "value", but this is
+/// not enforced
+///
+/// Map
+///   - child[0] entry: Struct
+///     - child[0] key: K
+///     - child[1] value: V
+///
+/// As a matter of convention, neither values of the "entry" nor "key" fields
```

Comment:
Good point; I think this is just a language issue. What I mean is that "neither entry nor key fields may have nulls"